viewing_party/party.py

```diff
@@ -1,23 +1,174 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+
+    #Checks if any of the dictionary keys are None
+    if title == None or genre == None or rating == None:
+        return None
+
+    movie = {
+        "title": title,
+        "genre": genre,
+        "rating": rating
+    }
+
+    return movie
+
+def add_to_watched(data_dict, movie):
+    #Adds the movie to the list value of the key watched
+
+    data_dict["watched"].append(movie)
+
+    return data_dict
+
+def add_to_watchlist(data_dict, movie):
+
+    data_dict["watchlist"].append(movie)
+
+    return data_dict
+
+def watch_movie(data_dict, movie_title):
+    # Locate list index of movie in dictionary watchlist
+    #index = 0
+    for movie in data_dict["watchlist"]:
+        if movie_title in movie["title"]:
+            data_dict["watched"].append(movie)
+            data_dict["watchlist"].remove(movie)
+    return data_dict
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    watched = user_data["watched"]
+    total_ratings = 0.0
+    
+    if not watched:
+        return 0.0
+    for movie in watched:
+        total_ratings += movie["rating"]
+    avg_rating = total_ratings/len(watched)
+    return avg_rating

```

Comment:
Consider adding a few blank lines within your function to separate logically distinct sections of the code (e.g., setup, processing, return).  According to [PEP 8](https://peps.python.org/pep-0008/#blank-lines), blank lines can improve readability by visually grouping related operations. Right now, everything runs together, which can make it harder to quickly understand the structure and flow of your function.

Operators like `/` should be surrounded with white spaces. Take a look at the [PEP8 style guide on white spaces](https://peps.python.org/pep-0008/#whitespace-in-expressions-and-statements).  In industry, code that doesn't follow an agreed upon code styles will need to be revised to ensure the codebase is neat and consistent.

```suggestion
    if not watched:
        return 0.0
        
    for movie in watched:
        total_ratings += movie["rating"]
        
    avg_rating = total_ratings / len(watched)
    return avg_rating
```